---
title: "General files"
author: "Lilly"
date: "2023-03-06"
output: html_document
---
```{r}
library(rtracklayer)
library(tidyverse)
library(gggenomes)
```

```{r}
## Load files (3 types)

#Load tyded gtf (csv)
bovis <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/bovis.csv")
becei <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/becei.csv")
panamensis <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/panamensis.csv")
inopinata <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/inopinata.csv")
elegans <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/elegans.csv")
tropicalis <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/tropicalis.csv")
remanei <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/remanei.csv")
latens <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/latens.csv")
tribulationis <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/tribulationis.csv")
briggsae <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/briggsae.csv")
nigoni <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/nigoni.csv")

#Load tyded psl files (csv)
psl_becei_bovis <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/becei_bovis.csv")
psl_bovis_becei <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/bovis_becei.csv")
psl_panamensis_becei <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/panamensis_becei.csv")
psl_becei_panamensis <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/becei_panamensis.csv")
psl_panamensis_inopinata <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/panamensis_inopinata.csv")
psl_inopinata_panamensis <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/inopinata_panamensis.csv")
psl_inopinata_elegans <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/inopinata_elegans.csv")
psl_elegans_inopinata <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/elegans_inopinata.csv")
psl_tropicalis_elegans <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/tropicalis_elegans.csv")
psl_elegans_tropicalis <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/elegans_tropicalis.csv")
psl_tropicalis_remanei <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/tropicalis_remanei.csv")
psl_remanei_tropicalis <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/remanei_tropicalis.csv")
psl_latens_remanei <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/latens_remanei.csv")
psl_remanei_latens <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/remanei_latens.csv")
psl_latens_tribulationis <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/latens_tribulationis.csv")
psl_tribulationis_latens <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/tribulationis_latens.psl")
psl_briggsae_tribulationis <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/briggsae_tribulationis.csv")
psl_tribulationis_briggsae <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/tribulationis_briggsae.csv")
psl_nigoni_briggsae <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/nigoni_briggsae.csv")
psl_briggsae_nigoni <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/briggsae_nigoni.csv")

#Load csv files of pairwise alignment with C.elegans 
genes_bovis <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/genes_bovis.csv")
genes_becei <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/genes_becei.csv")
genes_panamensis <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/genes_panamensis.csv")
genes_inopinata <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/genes_inopinata.csv")
genes_elegans <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/genes_elegans.csv")
genes_tropicalis  <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/genes_tropicalis.csv")
genes_remanei <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/genes_remanei.csv")
genes_latens <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/genes_latens.csv")
genes_tribulationis <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/genes_tribulationis.csv")
genes_briggsae <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/genes_briggsae.csv")
genes_nigoni <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/genes_nigoni.csv")


#Load psl pairwise alignment
psl_bovis <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/psl_bovis.csv")
psl_becei <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/psl_becei.csv")
psl_panamensis <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/psl_panamensis.csv")
psl_inopinata <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/psl_inopinata.csv")
psl_elegans <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/psl_elegans.csv")
psl_tropicalis  <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/psl_tropicalis.csv")
psl_remanei <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/psl_remanei.csv")
psl_latens <- read.csv("C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/psl_latens.csv")
psl_tribulationis <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/psl_tribulationis.csv")
psl_briggsae <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/psl_briggsae.csv")
psl_nigoni <- read.csv( "C:/Users/Lilly/OneDrive/Documents/Travail/Synteny_project/R_files/CSV/psl_nigoni.csv")

```

### Prepare data before the pipeline 

```{r}
## Clean it 
genes_becei <- genes_becei  %>% select(-X)
genes_bovis <- genes_bovis  %>% select(-X)
genes_panamensis <- genes_panamensis  %>% select(-X)
genes_inopinata <- genes_inopinata  %>% select(-X)
genes_elegans <- genes_elegans  %>% select(-X)
genes_tropicalis <- genes_tropicalis  %>% select(-X)
genes_remanei <- genes_remanei  %>% select(-X)
genes_latens <- genes_latens %>% select(-X)
genes_tribulationis <- genes_tribulationis  %>% select(-X)
genes_briggsae <- genes_briggsae  %>% select(-X)
genes_nigoni <- genes_nigoni  %>% select(-X)

psl_becei_bovis <- psl_becei_bovis  %>% select(-X)
psl_bovis_becei <- psl_bovis_becei  %>% select(-X)
psl_panamensis_becei <- psl_panamensis_becei  %>% select(-X)
psl_becei_panamensis <- psl_becei_panamensis  %>% select(-X)
psl_panamensis_inopinata <- psl_panamensis_inopinata  %>% select(-X)
psl_inopinata_panamensis <- psl_inopinata_panamensis  %>% select(-X)
psl_inopinata_elegans <- psl_inopinata_elegans  %>% select(-X)
psl_elegans_inopinata <- psl_elegans_inopinata  %>% select(-X)
psl_tropicalis_elegans <- psl_tropicalis_elegans  %>% select(-X)
psl_elegans_tropicalis <- psl_elegans_tropicalis  %>% select(-X)
psl_tropicalis_remanei <- psl_tropicalis_remanei  %>% select(-X)
psl_remanei_tropicalis <- psl_remanei_tropicalis  %>% select(-X)
psl_latens_remanei <- psl_latens_remanei  %>% select(-X)
psl_remanei_latens <- psl_remanei_latens  %>% select(-X)
psl_latens_tribulationis <- psl_latens_tribulationis  %>% select(-X)
psl_tribulationis_latens <- psl_tribulationis_latens  %>% select(-X)
psl_briggsae_tribulationis <- psl_briggsae_tribulationis  %>% select(-X)
psl_tribulationis_briggsae <-psl_tribulationis_briggsae  %>% select(-X)
psl_nigoni_briggsae <-psl_nigoni_briggsae  %>% select(-X)
psl_briggsae_nigoni <-psl_briggsae_nigoni  %>% select(-X)

psl_becei <- psl_becei  %>% select(-X)
psl_bovis <- psl_bovis  %>% select(-X)
psl_panamensis <- psl_panamensis  %>% select(-X)
psl_inopinata <- psl_inopinata  %>% select(-X)
psl_elegans <- psl_elegans  %>% select(-X)
psl_tropicalis <- psl_tropicalis  %>% select(-X)
psl_remanei <- psl_remanei  %>% select(-X)
psl_latens <- psl_latens %>% select(-X)
psl_tribulationis <- psl_tribulationis  %>% select(-X)
psl_briggsae <- psl_briggsae  %>% select(-X)
psl_nigoni <- psl_nigoni  %>% select(-X)
```

```{r}
## GTF genes files filter by type transcript 
filter_bovis <- bovis %>%  group_by(scaff) %>% filter(type == "transcript") %>% select(-X) %>% rename(length = "width")
filter_becei <- becei %>%  group_by(scaff) %>% filter(type == "transcript") %>% select(-X) %>% rename(length = "width")
filter_panamensis <- panamensis %>%  group_by(scaff) %>% filter(type == "transcript") %>% select(-X) %>% rename(length = "width")
filter_inopinata <- inopinata %>%  group_by(scaff) %>% filter(type == "transcript") %>% select(-X) %>% rename(length = "width")
filter_elegans <- elegans %>%  group_by(scaff) %>% filter(type == "transcript") %>% select(-X) %>% rename(length = "width")
filter_tropicalis <- tropicalis %>%  group_by(scaff) %>% filter(type == "transcript") %>% select(-X) %>% rename(length = "width")
filter_remanei <- remanei %>%  group_by(scaff) %>% filter(type == "transcript") %>% select(-X) %>% rename(length = "width")
filter_latens <- latens %>%  group_by(scaff) %>% filter(type == "transcript") %>% select(-X) %>% rename(length = "width")
filter_tribulationis <- tribulationis %>%  group_by(scaff) %>% filter(type == "transcript") %>% select(-X) %>% rename(length = "width")
filter_briggsae <- briggsae %>%  group_by(scaff) %>% filter(type == "transcript") %>% select(-X) %>% rename(length = "width")
filter_nigoni <- nigoni %>%  group_by(scaff) %>% filter(type == "transcript") %>% select(-X) %>% rename(length = "width")

# #Merge genes_transcripts by the filtered genes with C.elegans
transcript_bovis <- merge(filter_bovis, genes_bovis, by=c("start", "end"))
transcript_becei <- merge(filter_becei, genes_becei, by=c("start", "end"))
transcript_panamensis <- merge(filter_panamensis, genes_panamensis, by=c("start", "end"))
transcript_inopinata <- merge(filter_inopinata, genes_inopinata, by=c("start", "end"))
transcript_elegans <- merge(filter_elegans, genes_elegans, by=c("start", "end"))
transcript_tropicalis <- merge(filter_tropicalis, genes_tropicalis, by=c("start", "end"))
transcript_remanei <- merge(filter_remanei, genes_remanei, by=c("start", "end"))
transcript_latens <- merge(filter_latens, genes_latens, by=c("start", "end"))
transcript_tribulationis <- merge(filter_tribulationis, genes_tribulationis, by=c("start", "end"))
transcript_briggsae <- merge(filter_briggsae, genes_briggsae, by=c("start", "end"))
transcript_nigoni <- merge(filter_nigoni, genes_nigoni, by=c("start", "end"))

#rbind
inter <- rbind(transcript_bovis,transcript_becei,transcript_panamensis,transcript_inopinata,transcript_elegans,transcript_tropicalis,transcript_remanei,transcript_latens,transcript_tribulationis,transcript_briggsae,transcript_nigoni)

#Rename and relocate columns
genes <- inter %>%  rename(seq_id ="seq_id.x", scaff = "scaff.x", length = "length.x", strand = "strand.x", gene_id = "gene_id.x", scaff_elegans = "scaff.y", transcript_id_elegans = "gene_id.y") %>% select(-c("seq_id.y", "length.y", "strand.y")) %>% relocate(seq_id, .before = start) %>% relocate(scaff, .after = seq_id) %>% relocate(length, .after = end)
```


```{r}
#Get the transcript coordinates with min start and max end (we are building something that doesn't exist in the real life (macro synteny level))

#First version with summarise (remove the other informations)
min_max_genes <- genes %>% group_by(seq_id) %>% summarise(start = min(start), end = max(end)) %>% distinct()
min_max_genes <- merge(min_max_genes, genes, by = c("start", "end"), all.x = TRUE)

#Second version with mutate and ungroup (keep all the other columns (infos will be redundant in the new created columns))
min_max_genes <- genes %>% group_by(seq_id) %>% mutate(start_min = min(start), end_max = max(end)) %>% ungroup() %>% distinct()
```

```{r}
## Links
#Data frame with all the links following the tree
links_df <- rbind(psl_becei_bovis, psl_panamensis_becei, psl_inopinata_panamensis,psl_elegans_inopinata,psl_tropicalis_elegans,psl_remanei_tropicalis,psl_latens_remanei, psl_tribulationis_latens, psl_briggsae_tribulationis, psl_nigoni_briggsae, psl_bovis_becei, psl_becei_panamensis, psl_panamensis_inopinata, psl_inopinata_elegans, psl_elegans_tropicalis, psl_tropicalis_remanei, psl_remanei_latens, psl_latens_tribulationis, psl_tribulationis_briggsae, psl_briggsae_nigoni, psl_becei, psl_bovis, psl_tropicalis, psl_tribulationis, psl_panamensis, psl_remanei, psl_inopinata, psl_latens, psl_briggsae, psl_nigoni)
```

```{r}
# Save data 
write.csv(genes, "../R_files/CSV/genes.csv")
write.csv(min_max_genes, "../R_files/CSV/genes.csv")
writ.csv(links_df, "../R_files/CSV/links_df.csv")
```

-----------------------------------------------------------------PIPELINE--------------------------------------------------------------

### Pipeline with the general data 

```{r}
## Macro-synteny : MAIN
genes <- read.csv("../R_files/CSV/genes.csv")
min_max_genes <- read.csv("../R_files/CSV/min_max_genes.csv")
links_df <- read.csv("../R_files/CSV/links_df.csv")
```

```{r}
## User selection gene_id (examples)

```

