# WormSynteny

A tool to vizualize synteny among eleven species of nematodes taking Caenorhabditis Elegans as the query sequence.

I.	Introduction 

This project takes place after the work realised by Fuqiang Ma, Chun Yin Lau, and Chaogu Zheng about the Dynamic evolution of recently duplicated genes in Caenorhabditis elegans in the Zheng’s lab, at the University of Hong Kong. Some of the data generated in their study has been used to develop the application. 

This web application is aiming to fill the gap of application tools to visualize synteny between worms.  In current biology, synteny commonly refers to colinearity, i.e. conservation of blocks of order within two sets of chromosomes that are being compared with each other. That is, this application can be used to compare orthologs between some pre-selected nematodes. 

To do so, eleven species of nematodes have been selected from the gender Caenorhabditis: bovis, becei, panamensis, inopinata, elegans, tropicalis, remanei, latens, tribulationis, briggsae, and nigoni. The selection was based on the assembly contiguity (N50 > 200,000) and the completeness of genome assembly and annotation (BUSCO 428 assembly > 90%) from WORMBASE PARASITE website. 

Then, in order to get the orthogroups information and being able to compare the evolution of genes between elegans and those species, they extracted the longest protein isoform encoded by each gene in the eleven species using a clustering method such as the OrthoFinder v2.5.4.

Finally, they generated a phylogenetic tree inferred from all genes using the STAG algorithm and then rooted using the STRIDE algorithm for identifying gene duplication events. 
Genes with duplication events in more than one branch were assigned to the most recent one. Genes with duplication events within 60 million years were considered as recently duplicated genes. 

 ![image](https://github.com/LillyBouvarel/Synteny-Project/assets/79172540/472d6ca0-a9b3-4955-a690-7e42cd7b067c)

Duplication characteristics among eleven Caenorhabditis species. (A) The number of genes in the orthogroups (OGs) and genes duplicated at different times for each Caenorhabditis species identified by Duplication Loss Coalescence model.

II.	Data files format

a-	Genes information 

Genes data come from WormBase Parasite (GTF files format). 
The orthogroups information has been generated by Zheng’s lab (see introduction). 

To learn more about GTF file format click this link: https://genome.ucsc.edu/FAQ/FAQformat.html#format4.

b-	Alignment files

 Using GTF files, they first get the alignment between C.elegans and each species using the reference-free multiple genome alignment program Cactus. An HAL file format has been generated containing the alignment between the query and each target sequence. 

To learn more about HAL file format click this link: https://genome.ucsc.edu/FAQ/FAQformat.html#format4


III.	Application working logic.

To summarize, we have from now three types of files:
-	GTF files which provide information about genes coordinates and various genes metadata. 
-	HAL file which contains the Cactus alignment between the query sequence (C.elegans) and the targets sequences. 
-	CSV file which holds the orthogroups data for each gene

From here, we build CSV files to merge the gene_id and Orthogroups information in one unique file. 
Those files can be found in the Data repository. 

In the Ubuntu repository, you will find the HAL file as well as extra files generated while computing the synteny plots. 
