---
title: "All_infos_plots"
author: "Lilly"
date: "2023-02-09"
output: html_document
---

```{r, echo=FALSE}
library(gggenomes)
library(rtracklayer)
```


```{r, echo=FALSE}
#Import gtf files 
elegans <- rtracklayer::import("../synteny/caenorhabditis_elegans.PRJNA13758.WBPS16.canonical_geneset.gtf")
elegans=as.data.frame(elegans)

becei <- rtracklayer::import("../synteny/caenorhabditis_becei.PRJEB28243.WBPS16.canonical_geneset.gtf")
becei = as.data.frame(becei)

bovis <- rtracklayer::import("../synteny/caenorhabditis_bovis.PRJEB34497.WBPS16.canonical_geneset.gtf")
bovis=as.data.frame(bovis)

briggsae <- rtracklayer::import("../synteny/caenorhabditis_briggsae.PRJNA10731.WBPS16.canonical_geneset.gtf")
briggsae <- as.data.frame(briggsae)

inopinata <- rtracklayer::import("../synteny/caenorhabditis_inopinata.PRJDB5687.WBPS16.canonical_geneset.gtf")
inopinata <-  as.data.frame(inopinata)

latens <- rtracklayer::import("../synteny/caenorhabditis_latens.PRJNA248912.WBPS16.canonical_geneset.gtf")
latens <-  as.data.frame(latens)

nigoni <- rtracklayer::import("../synteny/caenorhabditis_nigoni.PRJNA384657.WBPS16.canonical_geneset.gtf")
nigoni <-  as.data.frame(nigoni)

panamensis <- rtracklayer::import("../synteny/caenorhabditis_panamensis.PRJEB28259.WBPS16.canonical_geneset.gtf")
panamensis <- as.data.frame(panamensis)

remanei <- rtracklayer::import("../synteny/caenorhabditis_remanei.PRJNA577507.WBPS16.canonical_geneset.gtf")
remanei <- as.data.frame(remanei)

tribulationis <- rtracklayer::import("../synteny/caenorhabditis_tribulationis.PRJEB12608.WBPS16.canonical_geneset.gtf")
tribulationis <- as.data.frame(tribulationis)

tropicalis <- rtracklayer::import("../synteny/caenorhabditis_tropicalis.PRJNA53597.WBPS16.canonical_geneset.gtf")
tropicalis <- as.data.frame(tropicalis)

#Import paf file
paf <- read.table("../synteny/evolver_Caenorhabditis.paf", sep = "\t")
```

```{r}
#Changes PAF colnames to match gggenomes requirements 
colnames(paf) <- c("seq_id", "length1", "start", "end", "strand", "seq_id2", "lenght2", "start2", "end2", "nmatch", "alen", "mapq", "other")

#Changes gtf files colname to match gggenomes requirements
colnames(elegans)[1] <- "scaff"
elegans$seq_id = paste("elegans", elegans$scaff ,sep = "_")

colnames(becei)[1] <- "scaff"
becei$seq_id = paste("becei", becei$scaff ,sep = "_")

colnames(bovis)[1] <- "scaff"
bovis$seq_id = paste("bovis", bovis$scaff ,sep = "_")

colnames(briggsae)[1] <- "scaff"
briggsae$seq_id = paste("briggsae", briggsae$scaff ,sep = "_")

colnames(inopinata)[1] <- "scaff"
inopinata$seq_id = paste("inopinata", inopinata$scaff ,sep = "_")

colnames(latens)[1] <- "scaff"
latens$seq_id = paste("latens", latens$scaff ,sep = "_")

colnames(nigoni)[1] <- "scaff"
nigoni$seq_id = paste("nigoni", nigoni$scaff ,sep = "_")

colnames(panamensis)[1] <- "scaff"
panamensis$seq_id = paste("panamensis", panamensis$scaff ,sep = "_")

colnames(remanei)[1] <- "scaff"
remanei$seq_id = paste("remanei", remanei$scaff ,sep = "_")

colnames(tribulationis)[1] <- "scaff"
tribulationis$seq_id = paste("tribulationis", tribulationis$scaff ,sep = "_")

colnames(tropicalis)[1] <- "scaff"
tropicalis$seq_id = paste("tropicalis", tropicalis$scaff ,sep = "_")
```


```{r}
## Read files 

#read python script prepared data
ava = read.table(file = "ele_X_1050000_1150000.nogap.ava")
seq = read.table(file = "ele_X_1050000_1150000.seq", header = T)
gene = read.table(file = "ele_X_1050000_1150000.nogap.ana")
```

```{r}
## Data frames manipulation (base to work with)

#For the seq : creating Seq length and seq_ID (length, seq_ID)
colnames(seq) = c("species", "scaff", "start", "end")
seq$length = as.numeric(seq$end) - as.numeric(seq$start) 
seq$seq_id = paste(seq$species, seq$scaff ,sep = "_")

#For the gene : creating seq_id, and rename the columns (seq_ID, start, end)
colnames(gene) = c("species", "scaff", "type", "start", "end", "strand","name")
gene$seq_id = paste(gene$species, gene$scaff ,sep = "_")
#gene$type[which(gene$type %in% "gene")] =  rep("CDS", 9) 
## if we want plot gene, only CDS region has color block

#For the links : we need seq_id, start, end and seq_id2, start2, end2 to link to sequences with each other
colnames(ava) = c("species1", "scaffold1", "start", "end", "strand", 
                  "species2", "scaffold2", "start2", "end2","strand2")
ava$seq_id = paste(ava$species1, ava$scaffold1, sep = "_")
ava$seq_id2 = paste(ava$species2, ava$scaffold2, sep = "_")

#To keep only the seq length that have the same seq_id that the sequences in the seq file 
ava$length = seq$length[match(ava$seq_id, seq$seq_id)]
ava$length2 = seq$length[match(ava$seq_id2, seq$seq_id)]

#Keep the useful columns (seq_id, length, start, end, strand, seq_id2, length2, start2, end2)
ava_sub = ava[ ,colnames(emale_ava)[2:10]] 
```

```{r}
#For the gene 
col <- c("scaff","start","end","width","strand","source","type","score","seq_id")
sub_elegans <- elegans[, col]
sub_elegans <- sub_elegans %>%  filter(scaff == "X")
sub_inopinata <- inopinata[, col]
sub_gene <- rbind(sub_elegans, sub_inopinata)

#Recreating the links file
sub_ava <- filter(ava, str_detect(ava$seq_id,"elegans|inopinata") & str_detect(ava$seq_id2,"elegans|inopinata"))
sub_ava_dup = sub_ava %>% filter(seq_id2 == seq_id)
sub_ava_no_dup = sub_ava %>% filter(seq_id2 != seq_id)
#unique(sub_gene$type)

#Geom_gene with gene_types == c(CDS,gene) and intron_types == gene
N4 <- gggenomes(genes = sub_elegans) + 
  geom_gene() +
  geom_seq() +
  geom_seq_label() 

N4  
```
```{r}
## Subsetting : N4 alignment (C.elegans and C.inopinata)
#For the gene 
col <- c("scaff","start","end","width","strand","source","type","score","seq_id")
colnames(elegans)
sub_elegans <- elegans[, col]
sub_inopinata <- inopinata[, col]
sub_gene_1 <- rbind(sub_elegans, sub_inopinata)

#subset data track (seq data) : get only C.elegans and C.inopinata 
elegans_inopinata <- seq %>% group_by(species) %>% filter(species %in% c("elegans","inopinata"))

#Recreating the other dataframe (gene and links)
sub_ava <- filter(ava, seq_id %in% elegans_inopinata$seq_id & seq_id2 %in% elegans_inopinata$seq_id)
sub_gene_2 <-  filter(sub_gene_1, seq_id %in% elegans_inopinata$seq_id)
sub_ava_dup <-  sub_ava %>% filter(seq_id2 == seq_id)
sub_ava_no_dup <-  sub_ava %>% filter(seq_id2 != seq_id)
#unique(sub_gene$type)

#Geom_gene with CDS only 
N4 <- gggenomes(genes = sub_gene_2) + 
  geom_gene(data=genes(.gene_types = c("start_codon", "gene")),intron_types = "gene") +
  geom_seq() +
  geom_seq_label()
N4
```



