---
title: "FirstAlignments"
author: "Lilly"
date: "2023-01-27"
output: html_document
---
```{r}
library(gggenomes)
```

```{r}
#Read files 
##read python script prepared data
ava = read.table(file = "ele_X_1050000_1150000.nogap.ava")
seq = read.table(file = "ele_X_1050000_1150000.seq", header = T)
gene = read.table(file = "ele_X_1050000_1150000.nogap.ana")
```

```{r}
#For the seq : creating Seq length and seq_ID (length, seq_ID)
colnames(seq) = c("species", "scaff", "start", "end")
seq$length = as.numeric(seq$end) - as.numeric(seq$start) 
seq$seq_id = paste(seq$species, seq$scaff ,sep = "_")

#For the gene : creating seq_id, and rename the columns (seq_ID, start, end)
colnames(gene) = c("species", "scaff", "type", "start", "end", "strand","name")
gene$seq_id = paste(gene$species, gene$scaff ,sep = "_")
#gene$type[which(gene$type %in% "gene")] =  rep("CDS", 9) 
## if we want plot gene, only CDS region has color block

#For the links : we need seq_id, start, end and seq_id2, start2, end2 to link to sequences with each other
colnames(ava) = c("species1", "scaffold1", "start", "end", "strand", 
                  "species2", "scaffold2", "start2", "end2","strand2")
ava$seq_id = paste(ava$species1, ava$scaffold1, sep = "_")
ava$seq_id2 = paste(ava$species2, ava$scaffold2, sep = "_")

#To keep only the seq length that have the same seq_id that the sequences in the seq file 
ava$length = seq$length[match(ava$seq_id, seq$seq_id)]
ava$length2 = seq$length[match(ava$seq_id2, seq$seq_id)]

#Keep the useful columns (seq_id, length, start, end, strand, seq_id2, length2, start2, end2)
ava_sub = ava[ ,colnames(emale_ava)[2:10]] 
```

```{r}
#With all the data : way too much (taking unraisonable time to run)
#p <- gggenomes(genes=gene, seqs=seq, links=ava_sub)
#p + 
  # geom_seq() +         # draw contig/chromosome lines
  # geom_seq_label() +   # label each sequence 
  # geom_gene() +        # draw genes as arrow
  # geom_link()          # draw some connections between syntenic regions

```
```{r}
##subset data track (seq data) : get the longest sequence per species 
by_species_single_seq <- seq %>% group_by(species) %>% 
  arrange(species, desc(length)) %>% 
  filter(row_number()==1)
```
```{r}
#Recreating the other dataframe (gene and links)
sub_ava <- filter(ava, seq_id %in% by_species_single_seq$seq_id & seq_id2 %in% by_species_single_seq$seq_id)
sub_gene = filter(gene, seq_id %in% by_species_single_seq$seq_id)
sub_ava_dup = sub_ava %>% filter(seq_id2 == seq_id)
sub_ava_no_dup = sub_ava %>% filter(seq_id2 != seq_id)
#unique(sub_gene$type)
```
```{r}
###gene dataset test
test_gene = sub_gene %>% group_by(species) %>% arrange(species, scaff, start) %>%
  filter(between(row_number(), 1,10))

#No differenciation between gene and CDS 
p <- gggenomes(genes = test_gene, links = sub_ava_no_dup) + 
  geom_gene() + 
  geom_seq()+
  geom_link()
p
```

```{r}
#Filter genes 
manipu_gene = sub_gene %>% filter(type == "gene")
p_gene <- gggenomes(genes = manipu_gene, links = sub_ava_no_dup) + 
  geom_gene() +
  geom_seq()+
  geom_link()+
  geom_seq_label()
p_gene
```




```{r}
#Filter CDS 
manipu_cds = sub_gene %>% filter(type == "CDS")
p_cds <- gggenomes(genes = manipu_cds, links = sub_ava_no_dup) + 
  geom_gene() + 
  geom_seq()+
  geom_link() +
  geom_seq_label()
p_cds
```



```{r}
#Need to create gggenomes tracks : seqs, genes, links, CDS knowing that geom_gene do not recognize the diff between cds and gene in column "type"
p_total <- gggenomes(genes = manipu_gene, links = sub_ava_no_dup, feats = list(manipu_cds)) + 
  geom_gene() + 
  geom_seq()+
  geom_link() +
  geom_seq_label()
p_total

```


```{r}
p %>% track_info()
p_gene %>% track_info()
p_cds %>% track_info()
p_total %>% track_info()

```

